File: client-sdk/example-usage.js
/**
 * Node.js/JavaScript client example for SathishLogger
 * Simple HTTP client to send logs to SathishLogger service
 */

const axios = require('axios');
const { v4: uuidv4 } = require('uuid');

class SathishLoggerClient {
    constructor(baseUrl, applicationName, apiKey = null) {
        this.baseUrl = baseUrl.replace(/\/$/, '');
        this.applicationName = applicationName;
        this.apiKey = apiKey;
        
        this.client = axios.create({
            timeout: 5000,
            headers: {
                'Content-Type': 'application/json',
                ...(apiKey && { 'X-API-Key': apiKey })
            }
        });
    }
    
    async info(message, correlationId = null, metadata = {}) {
        return this._log('INFO', message, correlationId, metadata);
    }
    
    async warn(message, correlationId = null, metadata = {}) {
        return this._log('WARN', message, correlationId, metadata);
    }
    
    async error(message, correlationId = null, error = null, metadata = {}) {
        return this._log('ERROR', message, correlationId, metadata, error);
    }
    
    async debug(message, correlationId = null, metadata = {}) {
        return this._log('DEBUG', message, correlationId, metadata);
    }
    
    async _log(level, message, correlationId = null, metadata = {}, error = null) {
        try {
            const logData = {
                applicationName: this.applicationName,
                logLevel: level,
                message: message,
                correlationId: correlationId || uuidv4(),
                timestamp: new Date().toISOString(),
                metadata: metadata
            };
            
            if (error) {
                logData.exceptionMessage = error.message;
                logData.stackTrace = error.stack;
            }
            
            const response = await this.client.post(`${this.baseUrl}/api/logs/log`, logData);
            return response.data;
            
        } catch (err) {
            // Fallback to console logging
            console.error(`Failed to send log to SathishLogger: ${err.message}`);
            console.log(`[${this.applicationName}] [${correlationId || 'N/A'}] [${level}] ${message}`);
        }
    }
    
    // Batch logging
    async logBatch(logs) {
        try {
            const response = await this.client.post(`${this.baseUrl}/api/logs/batch`, logs);
            return response.data;
        } catch (err) {
            console.error(`Failed to send batch logs to SathishLogger: ${err.message}`);
        }
    }
}

// Usage Example
async function main() {
    // Initialize logger
    const logger = new SathishLoggerClient('http://localhost:8080', 'nodejs-app');
    
    // Simple logging
    await logger.info('Node.js application started');
    await logger.warn('This is a warning message');
    
    // With correlation ID and metadata
    const correlationId = uuidv4();
    const metadata = { userId: '12345', requestPath: '/api/users' };
    await logger.info('Processing user request', correlationId, metadata);
    
    // Error logging with exception
    try {
        throw new Error('Something went wrong');
    } catch (error) {
        await logger.error('An error occurred', correlationId, error);
    }
    
    // Batch logging
    const batchLogs = [
        {
            applicationName: 'nodejs-app',
            logLevel: 'INFO',
            message: 'Batch log 1',
            correlationId: uuidv4(),
            timestamp: new Date().toISOString()
        },
        {
            applicationName: 'nodejs-app',
            logLevel: 'INFO',
            message: 'Batch log 2',
            correlationId: uuidv4(),
            timestamp: new Date().toISOString()
        }
    ];
    
    await logger.logBatch(batchLogs);
    
    console.log('Logs sent to SathishLogger service!');
}

// Run if this file is executed directly
if (require.main === module) {
    main().catch(console.error);
}

module.exports = SathishLoggerClient;

File: client-sdk/example-usage.py
#!/usr/bin/env python3
"""
Python client example for SathishLogger
Simple HTTP client to send logs to SathishLogger service
"""

import json
import requests
import uuid
from datetime import datetime
from typing import Optional, Dict, Any
import traceback

class SathishLoggerClient:
    def __init__(self, base_url: str, application_name: str, api_key: Optional[str] = None):
        self.base_url = base_url.rstrip('/')
        self.application_name = application_name
        self.api_key = api_key
        self.session = requests.Session()
        if api_key:
            self.session.headers.update({'X-API-Key': api_key})
    
    def info(self, message: str, correlation_id: Optional[str] = None, metadata: Optional[Dict[str, str]] = None):
        self._log('INFO', message, correlation_id, metadata)
    
    def warn(self, message: str, correlation_id: Optional[str] = None, metadata: Optional[Dict[str, str]] = None):
        self._log('WARN', message, correlation_id, metadata)
    
    def error(self, message: str, correlation_id: Optional[str] = None, exception: Optional[Exception] = None, metadata: Optional[Dict[str, str]] = None):
        self._log('ERROR', message, correlation_id, metadata, exception)
    
    def debug(self, message: str, correlation_id: Optional[str] = None, metadata: Optional[Dict[str, str]] = None):
        self._log('DEBUG', message, correlation_id, metadata)
    
    def _log(self, level: str, message: str, correlation_id: Optional[str] = None, 
             metadata: Optional[Dict[str, str]] = None, exception: Optional[Exception] = None):
        try:
            log_data = {
                'applicationName': self.application_name,
                'logLevel': level,
                'message': message,
                'correlationId': correlation_id or str(uuid.uuid4()),
                'timestamp': datetime.now().isoformat(),
                'metadata': metadata or {}
            }
            
            if exception:
                log_data['exceptionMessage'] = str(exception)
                log_data['stackTrace'] = traceback.format_exc()
            
            response = self.session.post(
                f'{self.base_url}/api/logs/log',
                json=log_data,
                timeout=5
            )
            response.raise_for_status()
            
        except Exception as e:
            # Fallback to console logging
            print(f"Failed to send log to SathishLogger: {e}")
            print(f"[{self.application_name}] [{correlation_id or 'N/A'}] [{level}] {message}")

# Usage Example
if __name__ == "__main__":
    # Initialize logger
    logger = SathishLoggerClient("http://localhost:8080", "python-app")
    
    # Simple logging
    logger.info("Python application started")
    logger.warn("This is a warning message")
    
    # With correlation ID and metadata
    correlation_id = str(uuid.uuid4())
    metadata = {"user_id": "12345", "request_path": "/api/users"}
    logger.info("Processing user request", correlation_id, metadata)
    
    # Error logging with exception
    try:
        raise ValueError("Something went wrong")
    except Exception as e:
        logger.error("An error occurred", correlation_id, e)
    
    print("Logs sent to SathishLogger service!")
